### PR TITLE
[MLOPS-580] Function schedule data endpoint

### PR DESCRIPTION
**Pull request by qTipTip**
# Some questions:

1. Should the SDK return `{'id' : 123, 'data' : {'key' : value}}`, or just
   `{'key' : value}`. I implemented the latter.
2. The mocking we use, what is that's main purpose? To avoid calling the API directly, we intercept the GET-request, and return a pre-configured JSON? Is this to only make
sure our SDK correctly calls the underlying API? 

# Commit messages

- Added a `get_data` method to the FunctionScheduleAPI.
- Added a `schedule_get_data_response` mock. What is the purpose of this? It intercepts the requests to the configured URL, and gathers statistics on the number of times called, etc?

### CODE REVIEW

File: cognite/experimental/_api/functions.py [hunk 1]
```diff
@@ -638,3 +638,25 @@ def delete(self, id: int) -> None:
         json = {"items": [{"id": id,}]}
         url = f"/functions/schedules/delete"
         self._post(url, json=json)
+
+    def get_data(self, id: int) -> Dict:
+        """
+        Get the data associated with a specific schedule.
```
Comment (HMEiding) [inline cognite/experimental/_api/functions.py:644]:
```suggestion
        Get the input data associated with a specific schedule.
```

**Review by HMEiding — CHANGES_REQUESTED** [review]
Nice! A couple of comments:
- It would be nice to also implement a method `get_data()` on the `FunctionSchedule` class. We've done the same for `get_response()` on the `FunctionCall` class.
- Update the CHANGELOG.md
- Bump the version.

**Comment by HMEiding** [comment]
> # Some questions:
> 1. Should the SDK return `{'id' : 123, 'data' : {'key' : value}}`, or just
>    `{'key' : value}`. I implemented the latter.
> 2. The mocking we use, what is that's main purpose? To avoid calling the API directly, we intercept the GET-request, and return a pre-configured JSON? Is this to only make
>    sure our SDK correctly calls the underlying API?
> 
> # Commit messages
> * Added a `get_data` method to the FunctionScheduleAPI.
> * Added a `schedule_get_data_response` mock. What is the purpose of this? It intercepts the requests to the configured URL, and gathers statistics on the number of times called, etc?

My take on your questions:

1. I think your solution is fine. This is in line with how we've done it for the `/response` endpoint.
2. Yes, with mocking we can test our code without calling the API directly. However, this means we will need to update the mocks whenever breaking changes occur in the API. Ideally, we should have integration tests as well, but we have so far not prioritized this.

File: pyproject.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "cognite-sdk-experimental"
-version = "0.31.1"
+version = "0.31.2"
```
Comment (HMEiding) [inline pyproject.toml:3]: Looking at this:
https://semver.org/
I think we should bump the minor version? So I was probably wrong in bumping the patch version last time...

Comment (qTipTip) [reply pyproject.toml]: Bumped to `0.32.0` instead :)

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,6 +12,11 @@ Changes are grouped as follows
 - `Fixed` for any bug fixes.
 - `Security` in case of vulnerabilities.
 
+## [0.33.0] - 2020-12-02
+### Added
+- Method `get_data` on the `FunctionSchedule`-class.
```
Comment (HMEiding) [inline CHANGELOG.md:17]: Update these two.

File: cognite/experimental/data_classes/functions.py [hunk 4]
```diff
@@ -166,6 +166,15 @@ def __init__(
         self.data = data
         self._cognite_client = cognite_client
 
+    def get_data(self) -> dict:
```
Comment (HMEiding) [inline cognite/experimental/data_classes/functions.py:169]: Should this also be `get_input_data()`?

File: cognite/experimental/data_classes/functions.py [hunk 5]
```diff
@@ -166,6 +166,15 @@ def __init__(
...
+        """
+        Get the input data associated to this schedule.
```
Comment (HMEiding) [inline cognite/experimental/data_classes/functions.py:171]: Update similarly to the API docs?

File: cognite/experimental/data_classes/functions.py [hunk 6]
```diff
@@ -166,6 +166,15 @@ def __init__(
...
+
+        Returns:
+            Input data associated with this schedule.
```
Comment (HMEiding) [inline cognite/experimental/data_classes/functions.py:174]: And here.

File: cognite/experimental/_api/functions.py [hunk 7]
```diff
@@ -638,3 +638,25 @@ def delete(self, id: int) -> None:
         json = {"items": [{"id": id,}]}
         url = f"/functions/schedules/delete"
         self._post(url, json=json)
+
+    def get_input_data(self, id: int) -> Dict:
+        """
+        Get the input data associated with a specific schedule.
```
Comment (HMEiding) [inline cognite/experimental/_api/functions.py:644]: Update similarly to the API docs?

**Review by HMEiding — CHANGES_REQUESTED** [review]
Som updates needed.

File: cognite/experimental/data_classes/functions.py (hunk 4 above)
Comment (qTipTip) [reply cognite/experimental/data_classes/functions.py]: Done!

File: cognite/experimental/data_classes/functions.py (hunk 6 above)
Comment (qTipTip) [reply cognite/experimental/data_classes/functions.py]: Done

File: cognite/experimental/_api/functions.py (hunk 7 above)
Comment (qTipTip) [reply cognite/experimental/_api/functions.py]: Done!

File: CHANGELOG.md [hunk 8]
```diff
@@ -12,6 +12,11 @@ Changes are grouped as follows
 - `Fixed` for any bug fixes.
 - `Security` in case of vulnerabilities.
 
+## [0.33.0] - 2020-12-02
+### Added
+- Method `get_input_data` on the `FunctionSchedule`-class. 
+- Method `get_input_data` on the `FunctionScheduleAPI`-class.
```
Comment (matzhaugen) [inline CHANGELOG.md:18]: Are both lines needed?

**Review by matzhaugen — APPROVED** [review]
Small nit but otherwise lgtm

**Review by HMEiding — APPROVED** [review]
LGTM 💯

**Pull request merged**